src/condor_utils/dagman_utils.cpp

```diff
@@ -1273,3 +1274,162 @@ std::string DagmanOptions::OptValueType(std::string& opt) {
+	const auto& [key, info] = *(dagOptionsInfoMap.lower_bound(flag));
+	if (strncasecmp(flag.c_str(), key.c_str(), flag.length()) != MATCH) { return ""; }
+	return key;
+}
+
+DagOptionInfo DagmanUtils::GetFlagInfo(const std::string& flag) { return DagmanGetFlagInfo(flag); }
+std::string DagmanUtils::GetFullFlag(const std::string& flag) { return DagmanGetFullFlag(flag); }
+
+void DagmanUtils::DisplayDAGManOptions(const char* fmt, DagOptionSrc source, const std::string opt_delim_meta) {
+	assert(fmt);
+	std::set<std::string> displayedOptions;
+	for (auto const& [flag, info] : dagOptionsInfoMap) {
+		const auto& [opt, metavar, desc, dispSrc] = info;
+
+		switch(source) {
+			case DagOptionSrc::DAGMAN_MAIN:
+				if ( ! (dispSrc & DAG_OPT_DISP_DAGMAN)) continue;
+				break;
+			case DagOptionSrc::CONDOR_SUBMIT_DAG:
+				if ( ! (dispSrc & DAG_OPT_DISP_CSD)) continue;
+				break;
+			case DagOptionSrc::PYTHON_BINDINGS:
+				if ( ! (dispSrc & DAG_OPT_DISP_PY_BIND)) continue;
+				if (displayedOptions.contains(opt)) continue;
+				displayedOptions.emplace(opt);
+				break;
+		}
+
+		std::string dispOpt = (source == DagOptionSrc::PYTHON_BINDINGS) ? opt : flag;
+		std::string rawType = "(" + DagmanOptValueType(opt) + ")";
+		if (rawType.find("bool") != std::string::npos) { rawType += "   "; }
+		if (rawType.find("string") != std::string::npos) { rawType += " "; }
+		std::string_view meta{""};
+		if ( ! IsOptionTypeBool(opt) || source == DagOptionSrc::PYTHON_BINDINGS) {
+			meta = metavar;
+			dispOpt += opt_delim_meta;
+		}
+
+		dispOpt += (source == DagOptionSrc::PYTHON_BINDINGS) ? rawType : meta;
+
+		fprintf(stdout, fmt, dispOpt.c_str(), desc.c_str());
+	}
+}
+
+std::string DagmanOptions::processOptionArg(const std::string& opt, std::string arg) {
+	if (strcasecmp(opt.c_str(), "AddToEnv") == MATCH) {
+		trim(arg); // Trim empty space around key=value pairs
+	} else if (strcasecmp(opt.c_str(), "BatchName") == MATCH) {
+		trim_quotes(arg, "\""); // Trim "" if any
+	}
+	return arg;
+}
+
+bool DagmanOptions::AutoParse(const std::string &flag, size_t &iArg, const size_t argc, const char * const argv[], std::string &err) {
+	SetDagOpt ret = SetDagOpt::KEY_DNE;
+	// Get information about flag
+	std::string fullFlag = DagmanGetFullFlag(flag);
+	const auto& [opt, meta, _, __] = DagmanGetFlagInfo(flag);
+	// No option means invalid flag
+	if (opt.empty()) {
+		formatstr(err,"Error: Unknown flag '%s' provided", flag.c_str());
+		return false;
+
+	// Handle bool options
+	} else if (IsOptionTypeBool(opt)) {
+		// Check if opposite flags for the same option were specified
+		if (boolFlagCheck.contains(opt)) {
+			std::string usedFlag = boolFlagCheck[opt];
+			if (usedFlag != fullFlag) {
+				formatstr(err, "Error: Both %s and %s can't be used at the same time",
+				          usedFlag.c_str(), fullFlag.c_str());
+				return false;
+			}
+		} else { boolFlagCheck.emplace(std::make_pair(opt, fullFlag)); }
+		// Parse with option infos metavar
```

Comment:
This appears to forbid `-boolflag true` or `-boolflag false`, but maybe DAGMan doesn't allow that?

Reply:
Conversation with ColeB confirmed that DAGMan requires all boolean options to be flags (that is, they can not have values).